### PR TITLE
move unmanaged nodegroup summaries to pkg/actions/nodegroup

### LINKED ISSUE #4736: Add unit tests for `GetUnmanagedNodeGroupSummaries`
`GetUnmanagedNodeGroupSummaries`, used by `get nodegroup`, has quite a lot of logic including listing CFN stacks for nodegroups, determining the nodegroup type from tags, and fetching the associated ASG and the ASG capacity values. But there are no unit tests for it, the existing unit tests mock `GetUnmanagedNodeGroupSummaries` itself, instead of the AWS API calls. 

Related to #4213

### PR DESCRIPTION
**Pull request by aclevername**
### Description
closes #4736

`GetUnmanagedNodeGroupSummaries`, used by `get nodegroup`, has quite a lot of logic including listing CFN stacks for nodegroups, determining the nodegroup type from tags, and fetching the associated ASG and the ASG capacity values. But there are no unit tests for it, the existing unit tests mock `GetUnmanagedNodeGroupSummaries` itself, instead of the AWS API calls. This PR moves this logic to `pkg/actions/nodegroup` and adds unit tests to it, as well as some minor refactoring

### CODE REVIEW

**Comment by Himangini** [comment]
Can you use the PR template and add some description, please?

File: pkg/actions/nodegroup/get.go [hunk 1]
```diff
@@ -58,27 +100,183 @@ func (m *Manager) GetAll() ([]*manager.NodeGroupSummary, error) {
...
 
-	if len(summaries) > 0 {
-		s := summaries[0]
-		if s.DesiredCapacity > 0 {
-			s.Version, err = kubewrapper.GetNodegroupKubernetesVersion(m.clientSet.CoreV1().Nodes(), s.Name)
-			if err != nil {
-				return nil, errors.Wrap(err, "getting nodegroup's kubernetes version")
-			}
+	// Create an empty array here so that an object is returned rather than null
+	summaries := []*Summary{}
```
Comment (Skarlso) [inline pkg/actions/nodegroup/get.go:110]: Uh, could you please use the make thingy instead? :D
```suggestion
	summaries := make([]*Summary, 0)
```

File: pkg/actions/nodegroup/get.go [hunk 2]
```diff
@@ -58,27 +100,183 @@ func (m *Manager) GetAll() ([]*manager.NodeGroupSummary, error) {
...
+	for _, s := range stacks {
+		summary, err := m.stackToSummary(s)
+		if err != nil {
+			return nil, err
+		}
+		if summary != nil {
```
Comment (Skarlso) [inline pkg/actions/nodegroup/get.go:116]: Don't need the nil check. Append doesn't do anything if it's nil.

File: pkg/cfn/manager/interface.go [hunk 3]
```diff
@@ -30,6 +31,8 @@ type GetNodegroupOption struct {
 	NodeGroupName string
 }
 
+var _ StackManager = &StackCollection{}
```
Comment (Skarlso) [inline pkg/cfn/manager/interface.go:34]: Noice

File: pkg/actions/nodegroup/get_test.go [hunk 4]
```diff
@@ -1,10 +1,12 @@
 package nodegroup_test
```
Comment (Skarlso) [inline pkg/actions/nodegroup/get_test.go:1]: Can we have at least some of the error scenarios tested too please? :) Pretty please? :D

**Review by Skarlso — COMMENTED** [review]
Just a few minor nits so far. Otherwise LGTM. Please re-tag me later.

File: pkg/actions/nodegroup/get.go (hunk 2 above)
Comment (aclevername) [reply pkg/actions/nodegroup/get.go:116]: i did not know that! wow

Comment (aclevername) [reply pkg/actions/nodegroup/get.go:116]: wait thats not true! https://go.dev/play/p/O4nfw0O-yn7

Comment (Skarlso) [reply pkg/actions/nodegroup/get.go:116]: Oh! So sorry, I misread this part. :( I thought it's a slice being added. Bah.

File: pkg/actions/nodegroup/get.go [hunk 5]
```diff
@@ -58,27 +100,183 @@ func (m *Manager) GetAll() ([]*manager.NodeGroupSummary, error) {
...
+	return m.stackToSummary(stack)
+}
+
+func (m *Manager) stackToSummary(s *manager.Stack) (*Summary, error) {
+	nodeGroupType, err := manager.GetNodeGroupType(s.Tags)
+	if err != nil {
+		return nil, err
+	}
+
+	if nodeGroupType != api.NodeGroupTypeUnmanaged {
```
[the remark this answers is not in the document]
Comment (aclevername) [reply pkg/actions/nodegroup/get.go:138]: image

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/actions/nodegroup/get.go:138]: I KNOW BUT THIS IS FRIGGING CONFUSING :D

[the remark this answers is not in the document]
Comment (aclevername) [reply pkg/actions/nodegroup/get.go]: actually we need this check, we call this and return `nil` to indicate its not an unmanaged node

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/actions/nodegroup/get.go:138]: uh :D

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/actions/nodegroup/get.go:138]: I knew  we probably call this elsewhere, the question was WHY if it returns. Just don't call it anywhere else. Anyways, that's out of scope. :)

**Review by Skarlso — APPROVED** [review]
>+614 −615

change something to make it `+615 −615`? :D

**Review by nikimanoledaki — APPROVED** [review]
read this 3 times and couldn't spot any issues. nice unit test refactoring! :) made the codebase a better place! 🌈

**Pull request merged**